src/lib/registration/migration_repositories.rb

```diff
@@ -0,0 +1,89 @@
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+#
+
+require "yast"
+require "registration/sw_mgmt"
+
+module Registration
+  # this class activates the migration repositories
+  class MigrationRepositories
+    include Yast::Logger
+
+    Yast.import "Pkg"
+
+    def self.reset
+      log.info "Resetting upgrade repos config"
+      repos = Yast::Pkg.GetUpgradeRepos()
+      repos.each { |repo| Yast::Pkg.RemoveUpgradeRepo(repo) }
+
+      # deselect all pre-selected packages or patches
+      Yast::Pkg.PkgReset
+    end
+
+    Yast.import "Pkg"
+
+    attr_accessor :repositories, :install_updates
+
+    def initialize
+      @repositories = []
+      # install updates by default
+      @install_updates = true
+    end
+
+    def add_service(service_name)
```

Comment:
missing documentation, especially for parameter, does it expect only string? and name as name or alias?